service/stop.go

```diff
@@ -22,20 +25,34 @@ func (service *Service) Stop() (err error) {
 
 // StopDependencies stops all dependencies
 func (service *Service) StopDependencies() (err error) {
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
 	for name, dependency := range service.GetDependencies() {
-		err = dependency.Stop(service.namespace(), name)
-		if err != nil {
-			return
-		}
+		wg.Add(1)
+		go func(d *Dependency, name string, dependencyName string) {
+			errStop := d.Stop(name, dependencyName)
+			mutex.Lock()
+			if errStop != nil && err == nil {
+				err = errStop
+			}
+			mutex.Unlock()
```

Comment:
same remark here about using `defer`

Reply:
also here i don't see why there is a mutex, in the map it makes sense because it's not thread safe if two threads want to write on the same index but with the error I don't think there is any problems

Reply:
Updated to use defer.

Reply:
The reason for the mutex is the `err` comparison in the `if`:
```go
if .... err == nil {
```